posts/2022-07-11-Rustup-1.25.0.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: post
+title: "Announcing Rustup 1.25.0"
+author: The Rustup Working Group
+---
+
+The rustup working group is happy to announce the release of rustup version 1.25.0. [Rustup][install] is the recommended tool to install [Rust][rust], a programming language that is empowering everyone to build reliable and efficient software.
+
+If you have a previous version of rustup installed, getting rustup 1.25.0 is as easy as closing your IDE and running:
```

Comment:
On Windows I can imagine that you wouldn't be able to update while any of the rustup shims is currently running. Closing your IDE reduces the risk of that. It's possible that rustup has some workaround for that Windows limitation though. By the way I did call Vim + rust-analyzer an IDE. As I understand it an IDE is an editor with the extra option to build your program from within the editor and present all the errors in a nice way, which rust-analyzer does by default on every change.

Reply:
@Mark-Simulacrum We started adding this note a few releases ago when it was evident that people were keeping their IDEs open in Windows, which meant we couldn't replace the proxies because Windows won't let you.  In general it's safer to say "Stop using rust before upgrading rustup" essentially.

Reply:
I think it would be good to add a footnote at least (or something like that) clarifying that this is more of a "in order to upgrade, we need to not have any of the proxies in use", rather than specifically about closing IDEs.

(And I'd guess that in most cases there's no actual need to close IDEs, since they're not persistently running any of rustup's proxies, so we'd be able to replace them in most cases?). But it seems OK to suggest it, I guess.

Reply:
I have tweaked this to "...is as easy as stopping any programs which may be using Rustup (e.g. closing your IDE) and running:"

Reply:
rust-analyzer, for example could choose to re-run cargo if you make an edit (even if you don't save it) which ends up using a proxy, so at least for Windows it's much safer to recommend people stop tools before running the update.

Reply:
I'm happy with this edit.